src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -1857,40 +1857,79 @@ protected String getArchiveEmail(String sessionId, String siteId) {
 		}
 
 		// Breakout the email name used in CTools for this archive.
-		protected String extractArchiveEmailName(String emailArchive) {
+		static public String extractArchiveEmailName(String emailArchive) {
+			String archive_email_name = null;
 			JSONObject jo = new JSONObject(emailArchive);
 			JSONArray ja = jo.getJSONArray("mailarchive_collection");
-			JSONArray firstHeaders = (JSONArray) ((JSONObject)ja.get(0)).get("headers");
-
-			int myJsonArraySize = firstHeaders.length();
-			String archive_email_name = null;
-
-			for (int i = 0; i < myJsonArraySize && archive_email_name == null; i++) {
-				String header  = (String) firstHeaders.get(i);
-				// look for the To: header
-				if (! header.startsWith("To: ")) {
-					continue;
+			outerloop:
+			for(int j=0;j<ja.length();j++) {
+				JSONArray allHeaders = (JSONArray) ((JSONObject) ja.get(j)).get("headers");
+				int myJsonArraySize = allHeaders.length();
+				for (int i = 0; i < myJsonArraySize && archive_email_name == null; i++) {
+					String header = (String) allHeaders.get(i);
+					// look for the To: header
+					if (!header.startsWith("To: ")) {
+						continue;
+					}
+					// Pull out the email name used by the old archive.
+					log.debug("aAEN: To header: {}", header);
+					archive_email_name = extractEmailFromToHeader(header);
+					if (archive_email_name == null) {
+						// breakout of inner for loop
+						break;
+					} else {
+						//found qualified google group name
+						break outerloop;
+					}
 				}
-				// Pull out the email name used by the old archive.
-				log.debug("aAEN: To header: {}",header);
-				archive_email_name = extractEmailFromToHeader(header);
-				break;
+			}
+			if(archive_email_name == null){
```

Comment:
Verify that this is ok with the BSA.  Log this at WARN level so that it is possible to go back and see what happened if an expected group doesn't show up after migration.  Since logs are temporary it would be even better to note this in the database somehow, but that may require more mechanism that we currently want to implement.